Test/Unit/Model/Api/DiscountCodeValidationTest.php

```diff
@@ -740,197 +550,1661 @@ public function validateWithIncorrectWebsiteId()
+        $result = [
+            'status'          => 'success',
+            'discount_code'   => self::COUPON_CODE,
+            'discount_amount' => $shippingDiscountAmount * 100,
+            'description'     => 'Discount ',
+            'discount_type'   => 'fixed_amount',
+            'cart'            => ['total_amount' => null, 'tax_amount' => null, 'discounts' => null]
+        ];
+
+        $this->request->expects(self::atLeastOnce())->method('getContent')
+            ->willReturn(json_encode($requestContent));
+
+        $this->shippingAddressMock->method('getDiscountAmount')->willReturn($shippingDiscountAmount);
+
+        $parentQuoteMock = $this->getQuoteMock(
+            self::COUPON_CODE,
+            null,
+            null,
+            false,
+            self::PARENT_QUOTE_ID,
+            self::PARENT_QUOTE_ID
+        );
+
+        $immutableQuoteMock = $this->getQuoteMock(self::COUPON_CODE);
+
+        $this->cartHelper->expects(self::once())->method('getActiveQuoteById')
+            ->with(self::PARENT_QUOTE_ID)->willReturn($parentQuoteMock);
+        $this->cartHelper->expects(self::once())->method('getQuoteById')
+            ->with(self::IMMUTABLE_QUOTE_ID)->willReturn($immutableQuoteMock);
+        $this->ruleMock->expects(self::once())->method('getSimpleAction')->willReturn('cart_fixed');
+
+        $this->couponMock->expects(self::atLeastOnce())->method('loadByCode')->willReturnSelf();
+        $this->couponMock->expects(self::atLeastOnce())->method('getCouponId')->willReturn(self::COUPON_ID);
+        $this->couponMock->expects(self::atLeastOnce())->method('getRuleId')->willReturn(self::RULE_ID);
+
+        $immutableQuoteMock->expects(self::once())->method('getItemsCount')->willReturn(1);
+        $parentQuoteMock->expects(self::atLeastOnce())->method('getCouponCode')->willReturn(self::COUPON_CODE);
+        $immutableQuoteMock->expects(self::atLeastOnce())->method('getCouponCode')->willReturn(self::COUPON_CODE);
+
+        $this->configHelper->expects(self::once())->method('getIgnoredShippingAddressCoupons')
+            ->willReturn([self::COUPON_CODE]);
+        $this->expectSuccessResponse($result);
+
+        self::assertTrue($this->currentMock->validate());
+    }
+
+    /**
+     * @test
+     */
+    public function validate_giftCardAccount()
+    {
+        $giftCardAmount = 15;
+        $requestContent = [
+            'cart' => [
+                'order_reference' => self::PARENT_QUOTE_ID,
+                'display_id'      => self::DISPLAY_ID,
+                'discount_code'   => self::COUPON_CODE
+            ]
+        ];
+        $result = [
+            'status'          => 'success',
+            'discount_code'   => self::COUPON_CODE,
+            'discount_amount' => $giftCardAmount * 100,
+            'description'     => 'Gift Card',
+            'discount_type'   => 'fixed_amount',
+            'cart'            => ['total_amount' => null, 'tax_amount' => null, 'discounts' => null]
```

Comment:
@danac-gs Can you clarify regarding `'discounts' => null`?
Why this field null if we have fields `discount_code` and `discount_amount` ?

Reply:
@YevhenShBolt this was passed to expectSuccessResponse which mocks the result. It could be anything and is irrelevant for this test. However I changed it to check for the expected values.